internal/unstruct/unstruct.go

```diff
@@ -0,0 +1,62 @@
+// unstruct has utilities for working with k8s unstructured data
+package unstruct
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/yaml"
+)
+
+// NestedMapSlice returns a copy of []map[string]interface{} value of a nested field.
+// Returns false if value is not found and an error if not a []interface{} or contains non-map items in the slice.
+// If the value is found but not of type []interface{}, this still returns true.
+func NestedMapSlice(obj map[string]interface{}, fields ...string) ([]map[string]interface{}, bool, error) {
```

Comment:
Since we're moving to Go 1.18, would it make sense to change `interface{}` -> `any`?